### DISCUSSION TITLE
Splitting up the GPhL extension file for easier approval

### DISCUSSION THREAD

**Discussion by pkeller** [pkeller/mmcif_pdbx#5] [General]
At the mmCIF WG subgroup on 2021-01-19, we discussed splitting up the single extension file so that it could be approved in sections. At the GPhL chat afterwards with @CV-GPhL and @GB-GPhL we agreed that we needed the following sections:

* anisotropic statistics (any name including "ellipsoid" or "aniso", and also '_reflns.gphl_orthogonalization_convention`)
* anomalous statistics (except those included in the anisotropic category)
* items to do with the per-reflection signal
* per-reflection multiplicity (which needs something to be done about AIMLESS's handling of centric reflections before we can actually implement this)

On closer examination, this doesn't cover everything:

* `_reflns.gphl_percent_possible_spherical_obs` and `_reflns_shell.gphl_percent_possible_spherical_all` can presumably be included in the anomalous statistics section (which we could call something else)
* the new items in the `diffrn_refln` category (unmerged reflections) don't fall into any of the above
* we also want to propose an application-level audit-style category 

These are my initial suggestions for the new extension filenames that are going to replace gphl-proc-extension.dic:

* gphl-proc-iso-stats-extension.dic
* gphl-proc-aniso-stats-extension.dic
* gphl-proc-unmerged-extension.dic
* gphl-proc-signal-extension.dic
* gphl-proc-refln-mult-extension.dic
* gphl-audit-extension.dic

This splitting up can't realistically be done until pull request #4 has been merged, except by dropping the pull request and applying the changes to the split files.

**Comment by pkeller** [comment]
Suggested split of the existing items in the GPhL extension:

### gphl-proc-iso-stats-extension.dic
 _reflns.gphl_percent_possible_spherical_obs
 _reflns.gphl_percent_possible_obs_anomalous
 _reflns.gphl_percent_possible_spherical_obs_anomalous
 _reflns.gphl_redundancy_anomalous
 _reflns.gphl_CC_half_anomalous
 _reflns.gphl_absDiff_over_sigma_anomalous

 _reflns_shell.gphl_percent_possible_spherical_all
 _reflns_shell.gphl_percent_possible_obs_anomalous
 _reflns_shell.gphl_percent_possible_spherical_obs_anomalous
 _reflns_shell.gphl_redundancy_anomalous
 _reflns_shell.gphl_CC_half_anomalous
 _reflns_shell.gphl_absDiff_over_sigma_anomalous

### gphl-proc-aniso-stats-extension.dic
 _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[1]
 _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[2]
 _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[3]
 _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[1]
 _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[2]
 _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[3]
 _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[1]
 _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[2]
 _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[3]
 _reflns.gphl_aniso_diffraction_limit_1
 _reflns.gphl_aniso_diffraction_limit_2
 _reflns.gphl_aniso_diffraction_limit_3
 _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[1]
 _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[2]
 _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[3]
 _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[1]
 _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[2]
 _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[3]
 _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[1]
 _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[2]
 _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[3]
 _reflns.gphl_aniso_B_tensor_eigenvalue_1
 _reflns.gphl_aniso_B_tensor_eigenvalue_2
 _reflns.gphl_aniso_B_tensor_eigenvalue_3
 _reflns.gphl_orthogonalization_convention

 _reflns.gphl_percent_possible_ellipsoidal_obs
 _reflns.gphl_percent_possible_ellipsoidal_obs_anomalous

 _reflns_shell.gphl_percent_possible_ellipsoidal_all
 _reflns_shell.gphl_percent_possible_ellipsoidal_obs_anomalous

### gphl-proc-unmerged-extension.dic
 _diffrn_refln.gphl_image_id
 _diffrn_refln.gphl_detector_x
 _diffrn_refln.gphl_detector_y
 _diffrn_refln.gphl_rotation_angle
 _diffrn_refln.gphl_scale_value

### gphl-proc-signal-extension.dic
 _reflns.gphl_observed_signal_threshold
 _reflns.gphl_signal_type
 _reflns.gphl_signal_details
 _reflns.gphl_signal_software_id

 _gphl_refln_signal_binning.ordinal
 _gphl_refln_signal_binning.upper_threshold
 
 _refln.gphl_signal
 _refln.gphl_signal_status
 
### gphl-proc-refln-mult-extension.dic
 _refln.gphl_number_obs
 _refln.gphl_number_obs_plus
 _refln.gphl_number_obs_minus

**Comment by CV-GPhL** [comment]
I would be inclined to split a little bit further with:

### gphl-proc-iso-stats-extension.dic
_reflns.gphl_percent_possible_spherical_obs
_reflns.gphl_percent_possible_spherical_obs_anomalous
_reflns_shell.gphl_percent_possible_spherical_all
_reflns_shell.gphl_percent_possible_spherical_obs_anomalous

### gphl-proc-anomalous-stats-extension.dic
_reflns.gphl_redundancy_anomalous
_reflns.gphl_CC_half_anomalous
_reflns.gphl_absDiff_over_sigma_anomalous
_reflns.gphl_percent_possible_obs_anomalous
_reflns_shell.gphl_redundancy_anomalous
_reflns_shell.gphl_CC_half_anomalous
_reflns_shell.gphl_absDiff_over_sigma_anomalous
_reflns_shell.gphl_percent_possible_obs_anomalous

and rename gphl-proc-aniso-stats-extension.dic to just gphl-proc-aniso-extension.dic (since it doesn't contain only stats). Does that make sense?

**Reply by pkeller** [reply]
OK, let's do that then.

**Reply by pkeller** [reply]
On second thoughts, I'm not sure about this, and it is starting to look to me as if the four `*_possible_spherical*` items should go into gphl-proc-aniso-extension.dic, since that is where they are meaningful.

**Reply by pkeller** [reply]
This is what I am about to commit:

### gphl-proc-aniso-extension.dic
  _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[1]
  _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[2]
  _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[3]
  _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[1]
  _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[2]
  _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[3]
  _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[1]
  _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[2]
  _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[3]
  _reflns.gphl_aniso_diffraction_limit_1
  _reflns.gphl_aniso_diffraction_limit_2
  _reflns.gphl_aniso_diffraction_limit_3
  _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[1]
  _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[2]
  _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[3]
  _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[1]
  _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[2]
  _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[3]
  _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[1]
  _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[2]
  _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[3]
  _reflns.gphl_aniso_B_tensor_eigenvalue_1
  _reflns.gphl_aniso_B_tensor_eigenvalue_2
  _reflns.gphl_aniso_B_tensor_eigenvalue_3
  _reflns.gphl_orthogonalization_convention
  _reflns.gphl_percent_possible_ellipsoidal_obs
  _reflns.gphl_percent_possible_spherical_obs
  _reflns.gphl_percent_possible_ellipsoidal_obs_anomalous
  _reflns.gphl_percent_possible_spherical_obs_anomalous
  _reflns_shell.gphl_percent_possible_ellipsoidal_all
  _reflns_shell.gphl_percent_possible_spherical_all
  _reflns_shell.gphl_percent_possible_ellipsoidal_obs_anomalous
  _reflns_shell.gphl_percent_possible_spherical_obs_anomalous
### gphl-proc-anomalous-stats-extension.dic
  _reflns.gphl_redundancy_anomalous
  _reflns.gphl_CC_half_anomalous
  _reflns.gphl_absDiff_over_sigma_anomalous
  _reflns.gphl_percent_possible_obs_anomalous
  _reflns_shell.gphl_redundancy_anomalous
  _reflns_shell.gphl_CC_half_anomalous
  _reflns_shell.gphl_absDiff_over_sigma_anomalous
  _reflns_shell.gphl_percent_possible_obs_anomalous
### gphl-proc-refln-mult-extension.dic
  _refln.gphl_number_obs
  _refln.gphl_number_obs_plus
  _refln.gphl_number_obs_minus
### gphl-proc-signal-extension.dic
  _reflns.gphl_observed_signal_threshold
  _reflns.gphl_signal_type
  _reflns.gphl_signal_details
  _reflns.gphl_signal_software_id
  _gphl_refln_signal_binning.ordinal
  _gphl_refln_signal_binning.upper_threshold
  _refln.gphl_signal
  _refln.gphl_signal_status
### gphl-proc-unmerged-extension.dic
  _diffrn_refln.gphl_image_id
  _diffrn_refln.gphl_detector_x
  _diffrn_refln.gphl_detector_y
  _diffrn_refln.gphl_rotation_angle
  _diffrn_refln.gphl_scale_value

**Reply by pkeller** [reply]
Committed: 764df5f

**Comment by CV-GPhL** [comment]
Hi Peter,

looks very good - there might be the odd renaming (_obs/_all
discussion), but the split seems to be the best we can do.

Thanks!

Clemens

On Wed, Jan 27, 2021 at 06:03:16AM -0800, Peter Keller wrote:
> This is what I am about to commit:
> 
> ### gphl-proc-aniso-extension.dic
>   _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[1]
>   _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[2]
>   _reflns.gphl_aniso_diffraction_limit_axis_1_ortho[3]
>   _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[1]
>   _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[2]
>   _reflns.gphl_aniso_diffraction_limit_axis_2_ortho[3]
>   _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[1]
>   _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[2]
>   _reflns.gphl_aniso_diffraction_limit_axis_3_ortho[3]
>   _reflns.gphl_aniso_diffraction_limit_1
>   _reflns.gphl_aniso_diffraction_limit_2
>   _reflns.gphl_aniso_diffraction_limit_3
>   _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[1]
>   _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[2]
>   _reflns.gphl_aniso_B_tensor_eigenvector_1_ortho[3]
>   _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[1]
>   _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[2]
>   _reflns.gphl_aniso_B_tensor_eigenvector_2_ortho[3]
>   _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[1]
>   _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[2]
>   _reflns.gphl_aniso_B_tensor_eigenvector_3_ortho[3]
>   _reflns.gphl_aniso_B_tensor_eigenvalue_1
>   _reflns.gphl_aniso_B_tensor_eigenvalue_2
>   _reflns.gphl_aniso_B_tensor_eigenvalue_3
>   _reflns.gphl_orthogonalization_convention
>   _reflns.gphl_percent_possible_ellipsoidal_obs
>   _reflns.gphl_percent_possible_spherical_obs
>   _reflns.gphl_percent_possible_ellipsoidal_obs_anomalous
>   _reflns.gphl_percent_possible_spherical_obs_anomalous
>   _reflns_shell.gphl_percent_possible_ellipsoidal_all
>   _reflns_shell.gphl_percent_possible_spherical_all
>   _reflns_shell.gphl_percent_possible_ellipsoidal_obs_anomalous
>   _reflns_shell.gphl_percent_possible_spherical_obs_anomalous
> ### gphl-proc-anomalous-stats-extension.dic
>   _reflns.gphl_redundancy_anomalous
>   _reflns.gphl_CC_half_anomalous
>   _reflns.gphl_absDiff_over_sigma_anomalous
>   _reflns.gphl_percent_possible_obs_anomalous
>   _reflns_shell.gphl_redundancy_anomalous
>   _reflns_shell.gphl_CC_half_anomalous
>   _reflns_shell.gphl_absDiff_over_sigma_anomalous
>   _reflns_shell.gphl_percent_possible_obs_anomalous
> ### gphl-proc-refln-mult-extension.dic
>   _refln.gphl_number_obs
>   _refln.gphl_number_obs_plus
>   _refln.gphl_number_obs_minus
> ### gphl-proc-signal-extension.dic
>   _reflns.gphl_observed_signal_threshold
>   _reflns.gphl_signal_type
>   _reflns.gphl_signal_details
>   _reflns.gphl_signal_software_id
>   _gphl_refln_signal_binning.ordinal
>   _gphl_refln_signal_binning.upper_threshold
>   _refln.gphl_signal
>   _refln.gphl_signal_status
> ### gphl-proc-unmerged-extension.dic
>   _diffrn_refln.gphl_image_id
>   _diffrn_refln.gphl_detector_x
>   _diffrn_refln.gphl_detector_y
>   _diffrn_refln.gphl_rotation_angle
>   _diffrn_refln.gphl_scale_value
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/pkeller/mmcif_pdbx/discussions/5#discussioncomment-314629

-- 

*--------------------------------------------------------------
* Clemens Vonrhein, Ph.D.     vonrhein AT GlobalPhasing DOT com
* Global Phasing Ltd., Sheraton House, Castle Park 
* Cambridge CB3 0AX, UK                   www.globalphasing.com
*--------------------------------------------------------------
